internal/controllers/clusterextension_controller.go

```diff
@@ -286,18 +294,16 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	managedObjs, _, err := r.Applier.Apply(ctx, unpackResult.Bundle, ext, objLbls, storeLbls)
 	if err != nil {
 		setStatusProgressing(ext, wrapErrorWithResolutionInfo(resolvedBundleMetadata, err))
-		// If bundle is not already installed, set Installed status condition to False
-		if installedBundle == nil {
-			setInstalledStatusConditionFailed(ext, err.Error())
-		}
+		// Now that we're actually trying to install, use the error
+		setInstalledStatusFromBundle(ext, installedBundle, err)
```

Comment:
>In this case, the Apply() failed. If we already had a bundle installed, we should continue to report that? But how do we report the error?

My understanding tracks with this continuing to be the `installedBundle` since we want to continue reporting that we successfully installed the previous bundle but are failing to upgrade. My understanding is that the error is reported through the progressing condition a bit higher up in this conditional.